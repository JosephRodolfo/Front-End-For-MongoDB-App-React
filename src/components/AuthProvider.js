import React from "react";
import { useState } from "react";

export const AuthContext = React.createContext(null);
export const useAuth = () => {

  return React.useContext(AuthContext);
};



const AuthProvider = ({ children }) => {


  const [token, setToken] = useState(null);

  const handleLogin = (user) => {
    setToken(user.token);
  };

  const handleLogout = () => {
    setToken(null);
  };

  const handleCreateUser = (user) =>{
    setToken(user.token)
  }

  const value = {
    token,
    onLogin: handleLogin,
    onLogout: handleLogout,
    onCreateUser: handleCreateUser,
  };

  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider> 

};

export default AuthProvider;
