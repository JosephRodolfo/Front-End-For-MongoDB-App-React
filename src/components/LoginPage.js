import React from "react";
import { startLogin } from "../actions/auth";
import { useState } from "react";

const LoginPage = ({ handleLogin }) => {
  const handleSubmit = async (e) => {
    e.preventDefault();
    const { username, password } = document.forms[0];

    startLogin(
      { email: username.value, password: password.value },
      handleLogin
    );
    //
  };

  const [loginData, setLoginData] = useState({
    email: "",
    password: "",
  });

  return (
    <div className="box-layout">
      <div className="box-layout__box">
        <form onSubmit={handleSubmit}>
          <h1 className="box-layout__title">Log in</h1>
          <input type="text" name="username" placeholder="email" />
          <input type="password" name="password" placeholder="password" />

          <button className="button" type="submit">
            Log in
          </button>
        </form>
      </div>
    </div>
  );
};

export default LoginPage;
