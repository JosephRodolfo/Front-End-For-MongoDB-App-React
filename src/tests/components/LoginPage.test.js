import LoginPage from "../../components/LoginPage";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import AuthProvider from "../../components/AuthProvider";
import "@testing-library/jest-dom";



let result;

beforeEach(() => {
  function Wrapper({ children }) {
    return (
      <BrowserRouter>
        <AuthProvider>{children}</AuthProvider>
      </BrowserRouter>
    );
  }

  result = render(<LoginPage />, { wrapper: Wrapper });
});

test("should render LoginPage correctly", () => {
  expect(result).toMatchSnapshot();

  expect(result.getByRole("button", { name: /Log in/i })).toBeInTheDocument();

  expect(result.getByRole("heading", { name: /Log in/i })).toBeInTheDocument();

  expect(
    result.getByRole("textbox", { name: /username/i })
  ).toBeInTheDocument();
});

