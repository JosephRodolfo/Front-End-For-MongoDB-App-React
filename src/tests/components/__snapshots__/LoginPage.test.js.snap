// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="box-layout"
      >
        <div
          class="box-layout__box"
        >
          <h1
            class="box-layout__title"
          >
            Log in
          </h1>
          <form
            class="form"
          >
            <div
              class="input-group"
            >
              <input
                aria-label="username"
                class=".input-group__item text-input"
                id="username"
                name="username"
                placeholder="email"
                type="text"
              />
              <input
                aria-label="password"
                class=".input-group__item text-input"
                id="password"
                name="password"
                placeholder="password"
                type="password"
              />
              <button
                class=".input-group__item button"
                type="submit"
              >
                Log in
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="box-layout"
    >
      <div
        class="box-layout__box"
      >
        <h1
          class="box-layout__title"
        >
          Log in
        </h1>
        <form
          class="form"
        >
          <div
            class="input-group"
          >
            <input
              aria-label="username"
              class=".input-group__item text-input"
              id="username"
              name="username"
              placeholder="email"
              type="text"
            />
            <input
              aria-label="password"
              class=".input-group__item text-input"
              id="password"
              name="password"
              placeholder="password"
              type="password"
            />
            <button
              class=".input-group__item button"
              type="submit"
            >
              Log in
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
